cpp/src/io/statistics/column_stats.cu

```diff
@@ -279,30 +276,25 @@ gatherFloatColumnStats(stats_state_s *s, statistics_dtype dtype, uint32_t t, Sto
   vmin = shuffle(vmin);
   vmax = warp_reduce(storage.float_stats[t / 32]).Reduce(vmax, cub::Max());
   vmax = shuffle(vmax);
-  vsum = warp_reduce(storage.float_stats[t / 32]).Reduce(vsum, IgnoreNaNSum());
   if (!(t & 0x1f)) {
     s->warp_min[t >> 5].fp_val = vmin;
     s->warp_max[t >> 5].fp_val = vmax;
-    s->warp_sum[t >> 5].fp_val = vsum;
   }
   has_minmax = __syncthreads_or(vmin <= vmax);
+  vsum       = block_reduce(storage.block_float_stats).Reduce(vsum, IgnoreNaNSum());
+  __syncthreads();
   if (t < 32 * 1) {
     vmin = warp_reduce(storage.float_stats[t / 32]).Reduce(s->warp_min[t].fp_val, cub::Min());
     if (!(t & 0x1f)) {
       s->ck.min_value.fp_val = (vmin != 0.0) ? vmin : CUDART_NEG_ZERO;
       s->ck.has_minmax       = (has_minmax);
+      s->ck.sum.fp_val       = vsum;
+      s->ck.has_sum          = (has_minmax);  // Implies sum is valid as well
```

Comment:
Any idea why parenthesis are used here?

Reply:
Nope, I had left it as it is from earlier code.